designs/add-recap-search-option-about-representation/readme.md

```diff
@@ -0,0 +1,113 @@
+**Proposed Change: Add a RECAP search option about representation.**
+
+*This proposal stems from freelawproject/courtlistener issue #[806](https://github.com/freelawproject/courtlistener/issues/806).*
+
+**CONTENTS**
+- Problem statement
+- Proposed solution by component
+  - User Interface
+  - Django code for parsing and storing docket data
+  - Solr
+  - Already-stored docket data
+  - Documentation
+- Test cases
+
+# Problem statement
+
+PACER and RECAP search users do not currently have an option to search for dockets where at least one petitioner had representation. Such dockets generally have higher quality complaints, motions, briefs, citations, and prose. Because that option is not available, searchers have to use other means that cost time and money to differentiate such cases from those where every petitioner is pro se.
+
+# Proposed solution 
+
+Enable a RECAP search option for retrieving dockets where at least one petitioner had representation.
+
+Enabling the option requires changes in (1) the user interface, (2) the server-side Django code that parses and stores docket data, (3) the Solr search configuration, (4) the already-stored docket data, and (5) documentation.
+
+The RECAP browser extension will not need to be changed.
+
+## Proposed solution: User Interface
+
+On the CourtListener.com [RECAP search page](https://www.courtlistener.com/recap/), **add** a select list to the search options.  The list's label will be: "At least one petitioner is represented".
+
+**Provide** three select list options.  The first empty/blank value will be the default.
+- ""
+- "Yes"
+- "No (Every party is pro se)"
```

Comment:
OK, so translating these a bit:

> Null — give me everything
> Yes — give me only high-quality cases
> No — give me only low-quality cases

Is there a reason to allow people to filter to cases where only pro se's are involved? That doesn't seem useful to me and isn't in the problem statement. I also have a lot of trouble (until I started writing this comment) understanding what the difference was between Null, Yes, and No.

What I'd like to really do is put a checkbox so this isn't a nullable boolean field. Something like:

 - [ ] Show only high-quality cases (?)

Or perhaps:

 - [ ] Hide pro se petitioner cases (?)

Then the (?) can explain what we mean. That seems a LOT simpler to understand/use than the more wordy "At least one petitioner is represented: Null, Yes, or No?" Is there a way to do this as a simple checkbox?

Reply:
How about "Exclude *pro se* petitioner cases (?)" ?   (I will commit a version with that change.)

>Is there a reason to allow people to filter to cases where only pro se's are involved?  That doesn't seem useful to me and isn't in the problem statement.

It won't be useful to me.  You're right:  it isn't in the problem statement.  I am anticipating that some researchers will want to focus on *pro se* cases only.  But until somebody says otherwise I won't provide that functionality through this change.  (At least not at the UI level.)

Reply:
That seems like a good way forward, yep.